pkg/terraform/diagnose.go

```diff
@@ -78,4 +78,9 @@ var conditions = []condition{{
 
 	reason:  "GCPComputeBackendTimeout",
 	message: `GCP is experiencing backend service interuptions, the compute instance failed to create in reasonable time.`,
+}, {
+	match: regexp.MustCompile(`Error: could not contact API: .*`),
```

Comment:
This regex match is a little too vague to not pick up non ironic failies imo.

Reply:
It currently matches two error messages that look like these:
```
level=error msg="Error: could not contact API: timeout reached"
level=error
level=error msg="  on ../../../../tmp/openshift-install-431515935/masters/main.tf line 1, in resource \"ironic_node_v1\" \"openshift-master-host\":"
level=error msg="   1: resource \"ironic_node_v1\" \"openshift-master-host\" {"
level=error
level=error
level=error
level=error msg="Error: could not contact API: timeout reached"
level=error
level=error msg="  on ../../../../tmp/openshift-install-431515935/masters/main.tf line 1, in resource \"ironic_node_v1\" \"openshift-master-host\":"
level=error msg="   1: resource \"ironic_node_v1\" \"openshift-master-host\" {"
level=error
level=error
level=error
level=error msg="Error: could not contact API: context deadline exceeded"
level=error
level=error msg="  on ../../../../tmp/openshift-install-431515935/masters/main.tf line 1, in resource \"ironic_node_v1\" \"openshift-master-host\":"
level=error msg="   1: resource \"ironic_node_v1\" \"openshift-master-host\" {"
level=error
```

Reply:
Yea, this is the maximum match. We can fix this in the terraform provider: https://github.com/openshift-metal3/terraform-provider-ironic/pull/45

Reply:
That's been merged and I released v0.2.3. You should be able to bump the go.mod

Reply:
Thanks. Won't we still have two different errors for the Ironic API errors? One from context and the other one we print explicitly?

Reply:
Added examples in the commit and PR messages.